src/mounted_mock.rs

```diff
@@ -46,7 +53,19 @@ impl MountedMock {
                 // Increase match count
                 self.n_matched_requests += 1;
                 // Keep track of request
-                self.matched_requests.push(request.clone())
+                self.matched_requests.push(request.clone());
+
+                // notification of satisfaction
+                if self
+                    .specification
+                    .expectation_range
+                    .contains(self.n_matched_requests)
```

Comment:
I'd avoid re-implementing verification logic here, we should delegate the check to the methods where it currently lives.